mock/contracts/esdtSimpleSC.go

```diff
@@ -40,3 +41,40 @@ func ExecESDTTransferAndCallParentMock(instanceMock *mock.InstanceMock, config i
 		return instance
 	})
 }
+
+// ExecESDTTransferWithAPICall is an exposed mock contract method
+func ExecESDTTransferWithAPICall(instanceMock *mock.InstanceMock, config interface{}) {
+	testConfig := config.(DirectCallGasTestConfig)
+	instanceMock.AddMockMethod("execESDTTransferWithAPICall", func() *mock.InstanceMock {
+		host := instanceMock.Host
+		instance := mock.GetMockInstance(host)
+		host.Metering().UseGas(testConfig.GasUsedByParent)
+
+		arguments := host.Runtime().Arguments()
+		if len(arguments) != 3 {
+			host.Runtime().SignalUserError("need 3 arguments")
+			return instance
+		}
+
+		input := test.DefaultTestContractCallInput()
+		input.CallerAddr = host.Runtime().GetSCAddress()
+		input.GasProvided = testConfig.GasProvidedToChild
+		input.Arguments = [][]byte{
+			test.ESDTTestTokenName,
+			big.NewInt(int64(testConfig.ESDTTokensToTransfer)).Bytes(),
+			arguments[2],
+		}
+		input.RecipientAddr = arguments[0]
+
+		elrondapi.TransferESDTNFTExecute(
+			host,
+			arguments[0],
+			test.ESDTTestTokenName,
+			testConfig.ESDTTokensToTransfer,
+			arguments[1],
+			[][]byte{arguments[2]},
```

Comment:
Should be defined as named variables, then passed as parameters, for readability.

Reply:
they are passed as parameters via `host.Runtime().Arguments()`